devdoc/tqueue_requirements.md

```diff
@@ -212,6 +247,26 @@ TQUEUE_PUSH_RESULT TQUEUE_PUSH(T)(TQUEUE(T) tqueue, T* item, void* copy_item_fun
 
 - **SRS_TQUEUE_01_022: [** If the queue is full (current head >= current tail + queue size), `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_QUEUE_FULL`. **]**
 
+- If the queue is full (current head >= current tail + queue size):
+
+  - If the queue is not growable, `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_QUEUE_FULL`.
+
+  - If the queue is growable:
+
+    - `TQUEUE_PUSH(T)` shall release in shared mode the lock used to guard the growing of the queue.
+
+    - `TQUEUE_PUSH(T)` shall acquire in exclusive mode the lock used to guard the growing of the queue.
+
+    - `TQUEUE_PUSH(T)` shall double the size of the queue.
```

Comment:
hmm, acquire-release-acquire sounds racy, likely needs another check in case a different thread already grew it between the release and the acquire.

Reply:
Added an extra spec about it.